view/next-project/src/pages/purchaseorders/index.tsx

```diff
@@ -170,7 +178,7 @@ export default function PurchaseOrders(props: Props) {
               </tr>
             </thead>
             <tbody className='border border-x-white-0 border-b-primary-1 border-t-white-0'>
-              {purchaseOrderViews.map((purchaseOrderViewItem, index) => (
+              {purchaseOrderViews.filter((purchaseOrderViewItem) => (purchaseOrderViewItem.purchaseOrder.deadline.includes(String(selectedYear.year)))).map((purchaseOrderViewItem, index) => (
```

Comment:
できればここは、動的に毎回filterしてmapするんじゃなく、useMemoを使って状態管理したい！
useMemoを使うことでメモ化して、値が違う時だけレンダリングすることができる

selectedYearを宣言したあとあたりで
```
  const filteredPurchaseOrderViews = useMemo(() => {
    return purchaseOrderViews.filter((purchaseOrderView: PurchaseOrderView) => {
      return purchaseOrderView.purchaseOrder.deadline.includes(selectedYear);
    });
  }, [purchaseOrderViews, selectedYear]);
```
みたいに、filter済みのものを作っておくと、ここは
```suggestion
              {filteredPurchaseOrderViews.map((purchaseOrderViewItem, index) => (

```
にすることができる！